app/discussion/client/views/DiscussionTabbar.js

```diff
@@ -24,24 +28,23 @@ Template.discussionsTabbar.helpers({
 
 Template.discussionsTabbar.onCreated(function() {
 	this.rid = this.data.rid;
-	this.cursor = DiscussionOfRoom.find({
-		rid: this.rid,
-	}, {
-		sort: {
-			ts: -1,
-		},
-	});
 	this.hasMore = new ReactiveVar(true);
 	this.limit = new ReactiveVar(50);
-	return this.autorun(() => {
+	this.cursor = new ReactiveVar([]);
+	this.autorun(() => {
 		const data = Template.currentData();
-		return this.subscribe('discussionsOfRoom', data.rid, this.limit.get(), () => {
-			if (DiscussionOfRoom.find({
+		const handle = Meteor.subscribe('discussionsOfRoom', data.rid, this.limit.get());
+		const isReady = handle.ready();
+		// Wait for collection to be ready
+		if (isReady) {
+			const discussions = DiscussionOfRoom.find({
 				rid: data.rid,
-			}).count() < this.limit.get()) {
+			}).fetch();
+			this.cursor.set(discussions);
```

Comment:
its now how cursor works, if you use a fetch()... its the data itself and not a cursor (:

Reply:
So I should try with cursors only?

Reply:
i get it cause we don't want to load all of them into your server's RAM. Let me fix with cursors